### ISSUE TITLE
Idea: support CSV files

### ISSUE THREAD

**Issue by tacman** [fire015/flintstone#52]
I have an idea for a Flintstone CSV databaseuse.

Why CSV?  

* Supported by php via fgetcsv, so very fast
* Supports multi-line automatically, no conversion
* Usable by other applications, csv is universal
* smallest format, since keys are not repeated.

Obviously, this would require that the keys be defined ahead of time, since of at runtime.  So it might not be the right solution for everyone.

I've written a class that returns an associative array from a CSV file, I use it as a generic way of loading CVS.

https://github.com/survos/survos/blob/main/packages/grid-group-bundle/src/Service/Reader.php

For example, I want to save the results of my API calls to a translation service, and want to store the source language, target languate, original text and translated text.

I can imagine it working like this

```php
        $transCache = new CsvFlintstone($prefix, [
            'key' => '_id', // the lookup key, could be default
            'keys' => ['source','target','orginal', 'translated_text'],


        $translation = $translationService($source, $textToTranslate);
        $key = md5($textToTranslate . $source);

          $transCache->set($key, array_merge(['source' => $source, 'target' => $target, 
                   'original' => $textToTranslate,                  
                    'translated_text' => $translation]));
```

This would create a CSV file
```csv
_id,source,target,original,translated_text
294FD2412,en,es,hello,hola
4935535FD,en,fr,hello,bon jour
```

It's small, fast, and re-usable.

I think it's just a matter of extending the Flintstone database class.  Thoughts?

**Comment by tacman** [comment]
I extended the class and got it working to use CSV as the underlying structure in my bundle.  I'm pretty pleased with it for my needs.  

I also added an intermediate cache that stores the key and offset to the database, so instead of looping through the lines it checks the offset cache, and does an fseek() to the correct position.

A contains() method simply returns if the key is in the offset cache, which is much faster.

The code for all this is here: https://github.com/survos/SurvosGridGroupBundle/tree/main/src/Service

Next I'd like to incorporate this as a Symfony Cache Adapter and have the component be able to fulfill a cache contract. 
 (https://symfony.com/doc/current/components/cache.html#available-cache-adapters).


Is this something of interest to you, @fire015, to incorporate here, maybe as a Version 2?  Or does it deviate enough from this library to be its own component?
